FWCore/Utilities/interface/ResourceInformation.h

```diff
@@ -0,0 +1,51 @@
+#ifndef FWCore_Utilities_ResourceInformation_h
+#define FWCore_Utilities_ResourceInformation_h
+
+/** \class edm::ResourceInformation
+
+  Description:
+
+  Usage:
+
+\author W. David Dagenhart, created May 3, 2022
+*/
+
+#include <string>
+#include <vector>
+
+namespace edm {
+
+  class ResourceInformation {
+  public:
+    ResourceInformation();
+    ResourceInformation(ResourceInformation const&) = delete;
+    ResourceInformation const& operator=(ResourceInformation const&) = delete;
+    virtual ~ResourceInformation();
+
+    enum class AcceleratorType { GPU };
```

Comment:
https://github.com/cms-sw/cmssw/pull/39402 made me think if knowing the GPU vendor (or "software stack"?) would be useful as well. In principle the information is available via `gpuModels()` (needs string parsing) or e.g. `nvidiaDriverVersion()` (check if the string is empty), so it could be checked already with the present interface. So maybe the present interface is sufficient, and we adjust later if it seems useful?